lib/booking.rb

```diff
@@ -22,23 +22,24 @@ def self.create(listing_id:, visitor_id:, status: 'pending')
 
   def self.all
     results = DatabaseConnection.query("SELECT * FROM booking order by id desc")
-    results.map do |row|
-      Booking.new(id: row['id'],
-                  visitor_id: row['visitor_id'],
-                  listing_id: row['listing_id'],
-                  status: row['status'])
-                end
+    Booking.map_results(results)
   end
 
-  def self.find_all(by:, id:)
+  def self.where(field:, id:)
     return nil unless id
-    return nil unless by == "visitor" || by == "listing"
-    results = DatabaseConnection.query("SELECT * FROM booking WHERE #{by}_id = '#{id}';")
+    return nil unless field == "visitor" || field == "listing"
+    results = DatabaseConnection.query("SELECT * FROM booking WHERE #{field}_id = '#{id}';")
+    Booking.map_results(results)
+  end
+
+  private
+
+  def self.map_results(results)
     results.map do |row|
       Booking.new(id: row['id'],
                   visitor_id: row['visitor_id'],
                   listing_id: row['listing_id'],
                   status: row['status'])
-                end
+    end
```

Comment:
Just a little nugget from hugh's brain: if you want to make a class method private, just putting private before it doesn't work. You write it normally but then afterwards you put private_class_method :map_results